### PR TITLE
Allow Intune Compliance Check to handle InGracePeriod

### PR DESCRIPTION
**Pull request by PureOxygen-Dev**
# Description
(REQUIRED)
Describe what the new code is used for.
The current intune compliance checker does not handle the InGracePeriod returned by the Graph API.

This request added a check on the condition and if there returns as compliant.  

In addition, it gives the user the option in the provisioner setup form to select a targeted security event

# Impacts
(REQUIRED)
Describe what to reviewer should look at. Will also help for testing purposes.
Provisioner setup form and Intune Compliance Check

### CODE REVIEW

File: lib/pf/provisioner/intune.pm [hunk 1]
```diff
@@ -294,7 +293,12 @@ sub verify_compliance {
                 if ($entry->{complianceState} eq 'compliant') {
                     $logger->info("Device $mac is compliant.");
                     return $TRUE;
-                } else {
+                } 
+                elsif ($entry->{complianceState} eq 'inGracePeriod') {
+					$logger->info("Device $mac is InGracePeriod.");
```
Comment (fdurand) [inline lib/pf/provisioner/intune.pm]: incorrect indentation

Comment (PureOxygen-Dev) [reply lib/pf/provisioner/intune.pm]: Hi Fabrice, sorry I missed in the developer guide about tabs.  Converted to spaces.

**Comment by PureOxygen-Dev** [comment]
@fdurand Hang on this one a little more please.  I think the whole intune compliance part is not working as expected and I am dropping a message to @julsemaan on the subject.

**Comment by PureOxygen-Dev** [comment]
Based on what this feature does I would only accept the compliant value back to trigger the captive portal form.  InGracePeriod could be misleading and result in the incorrect end result.

**Comment by mellalahmed** [comment]
Can you help me integrate security events in PacketFence?

**Pull request closed without merging**